Access control: enable RBAC by default

IevaVasiljeva opened the pull request:
**What this PR does / why we need it**:
Enables RBAC by default:
* add a new configuration section for RBAC in Grafana config;
* remove RBAC feature toggle;
* fix some tests that require a bit more extra setup with RBAC.

**Which issue(s) this PR fixes**:

Related to https://github.com/grafana/grafana-enterprise/issues/2503

**Special notes for your reviewer**:
related enterprise PR: https://github.com/grafana/grafana-enterprise/pull/3262

IevaVasiljeva committed:
cherry pick initial commit

IevaVasiljeva committed:
remove toggle

(cherry picked from commit 80c487d854312718afe9cdf0249c73a1cbd97517)

IevaVasiljeva committed:
frontend changes

(cherry picked from commit 34751e8e9367858afab8658323c2fa3701c721c8)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/grafana/grafana?pullRequest=48719) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/grafana/grafana?pullRequest=48719) before we can accept your contribution.<br/>**33** out of **34** committers have signed the CLA.<br/><br/>:white_check_mark: joshhunt<br/>:white_check_mark: ashharrison90<br/>:white_check_mark: ryantxu<br/>:white_check_mark: joey-grafana<br/>:white_check_mark: jalevin<br/>:white_check_mark: adela-almasan<br/>:white_check_mark: nmarrs<br/>:white_check_mark: wbrowne<br/>:white_check_mark: kaydelaney<br/>:white_check_mark: zoltanbedi<br/>:white_check_mark: gillesdemey<br/>:white_check_mark: sunker<br/>:white_check_mark: krzysdabro<br/>:white_check_mark: grafanabot<br/>:white_check_mark: gabor<br/>:white_check_mark: JoaoSilvaGrafana<br/>:white_check_mark: glintik<br/>:white_check_mark: yuri-tceretian<br/>:white_check_mark: asimpson<br/>:white_check_mark: achatterjee-grafana<br/>:white_check_mark: torkelo<br/>:white_check_mark: gotjosh<br/>:white_check_mark: IevaVasiljeva<br/>:white_check_mark: axelavargas<br/>:white_check_mark: josiahg<br/>:white_check_mark: toddtreece<br/>:white_check_mark: ying-jeanne<br/>:white_check_mark: kalleep<br/>:white_check_mark: L-M-K-B<br/>:white_check_mark: Elfo404<br/>:white_check_mark: marefr<br/>:white_check_mark: jalopez<br/>:white_check_mark: idafurjes<br/>:x: renovate[bot]<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/grafana/grafana?pullRequest=48719) it.</sub>

Jguer committed:
AccessControl: don't pull builtin role assignments when refactor is enabled (#48675)

gabor committed:
Loki: backend: use streaming JSON parser, try2 (#48752)

* converter: remove __name__ customization because Loki does not do that

Loki does not handle __name__ in a special way.
for Prometheus, the caller can implement the formatting
by themselves

* converter: change labels-formatting

the labels.String() method does not handle strange values well

* loki: backend: use streaming-json parser

* more idiomatic code

Co-authored-by: Todd Treece <360020+toddtreece@users.noreply.github.com>

* simpler row-length check

* simpler code

* fixed converter/prom tests

Co-authored-by: Todd Treece <360020+toddtreece@users.noreply.github.com>

dsotirakis committed:
Remove folder (#48617)

konrad147 committed:
Alerting: Improved RBAC for Alert managers (#48344)

* Initial support for grafana or cloud only alert managers

* Handle missing alert manager

* Refactor code, fix tests

* Fix redirect url

* Bring back the test

* Improve missing alert manager warning, add useAlertManagerSourceName tests

* Fix lint errors

* Rename alert manager hook

* Refactor alert manager label creation

* Improve warnings' messages

* Fix linter

* Fix warning condition in RuleEditor

papagian committed:
Alerting: modify ruler endpoints for proxying using the datasource UID (#48046)

* Modify ruler endpoints to expect the data source UID

* Update frontend

* Apply suggestion from code review

fridgepoet committed:
CloudWatch: Pass label in GetMetricData API request when dynamic label feature toggle is enabled (#48574)

connorlindsey committed:
Trace to logs: Only show loki and splunk datasources in settings (#48723)

joshhunt committed:
Tooltip: Make tooltip use secondary background color for legible links (#48748)

joshhunt committed:
Chore: Generate JSON theme files (#48762)

L-M-K-B committed:
Laura/chore/refactor test of secondary actions (#48745)

* Chore: transfer three tests to testing-library

* Chore: transfer last test to testing-library

* chore: add suggestion from code review

Co-authored-by: Piotr Jamróz <pm.jamroz@gmail.com>

Co-authored-by: Piotr Jamróz <pm.jamroz@gmail.com>

joshhunt committed:
QueryEditor: Set data source type in mixed query data source ref (#48734)

ivanahuckova committed:
Elasticsearch: Remove support for versions after their end of the life (<7.10.0) (#48715)

* Elasticsearch: Remove support for versions after their EOL

* Update docs

* Remove old versions from config

* Update pkg/tsdb/elasticsearch/elasticsearch.go

Co-authored-by: Gábor Farkas <gabor.farkas@gmail.com>

* Fix tests

* Fix typecheck errors

Co-authored-by: Gábor Farkas <gabor.farkas@gmail.com>

IevaVasiljeva commented:
I did a bad rebase and messed this PR up, sorry all! I'll close this one and open a clean one.
